### PR TITLE
Added support for query parameters

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

File: riptide-core/src/main/java/org/zalando/riptide/Requester.java [hunk 1]
```diff
@@ -30,8 +32,14 @@
 
 public abstract class Requester extends Dispatcher {
 
+    private final Multimap<String, String> query = LinkedHashMultimap.create();
     private final HttpHeaders headers = new HttpHeaders();
 
+    public final Requester query(final String name, final String value) {
```
Comment (AlexanderYastrebov) [inline riptide-core/src/main/java/org/zalando/riptide/Requester.java]: `param`?

**Comment by AlexanderYastrebov** [comment]
I thought UrlTemplate does this substitution

File: riptide-core/src/main/java/org/zalando/riptide/Requester.java (hunk 1 above)
Comment (lukasniemeier-zalando) [inline riptide-core/src/main/java/org/zalando/riptide/Requester.java]: `queryParam`?

Comment (whiskeysierra) [inline riptide-core/src/main/java/org/zalando/riptide/Requester.java]: Yes, if we change it then to `queryParam`. `param` itself has too many possible meanings in the context of HTTP: query param, post param, form param, ...

**Comment by whiskeysierra** [comment]
> I thought UrlTemplate does this substitution

No, UriTemplate is only doing parameter replacement, but it doesn't have support for query parameters as far as I can tell.

**Comment by tkrop** [comment]
@whiskeysierra you should try it. DefaultUriTemplateHandler does substitute query parameter too.
Try it out:
```
DefaultUriTemplateHandler h = new DefaultUriTemplateHandler();
        h.setBaseUrl("http://xample.com");
        URI uri = h.expand("/path/{a}/{b}/x?y={c}&z={d}", "r", "s", "t", "u");
        System.out.println(uri);
        uri = h.expand("/path/{a}/{b}/x?y={c}&z={d}",
                (Map<String,?>) ImmutableMap.<String, Object>builder().put("b", "r").put("a", "s").put("d", "t").put("c", "u").build());
        System.out.println(uri);
```

So this feature is mostly redundant to the uri-template support. In my opinion, the only use-case with additional value would require the interface `queryParams(final Multimap<String, String> query)`, that allows to ad any number of query parameters defined by the client.

**Comment by whiskeysierra** [comment]
Yes, that works but It doesn't feel very natural to me, and the overload makes sense.

**Comment by lukasniemeier-zalando** [comment]
*As* a riptide user
*I do not want* to fiddle with `UriTemplateHandler` in most cases
*So that* I build my URL in one place

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: riptide-core/src/main/java/org/zalando/riptide/Rest.java [hunk 2]
```diff
@@ -120,30 +124,36 @@ public final Requester trace(final URI url) {
 
     private Requester execute(final HttpMethod method, final String uriTemplate,
             final Object... uriVariables) {
-        return execute(method, uriTemplateHandler.expand(uriTemplate, uriVariables));
+        return new ListenableFutureRequester(method, fromUriString(uriTemplate), uriVariables);
```
Comment (tkrop) [inline riptide-core/src/main/java/org/zalando/riptide/Rest.java]: The naming is bad. Actually, I have the feeling the methods of this class should be integrated into the Requester itself.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**